src/mui/list/Export.js

```diff
@@ -0,0 +1,108 @@
+import React, { PropTypes, Component } from 'react';
+import LinearProgress from 'material-ui/LinearProgress';
+import ReactDOMServer from 'react-dom/server';
+import debounce from 'lodash.debounce';
+import santizeHtml from 'sanitize-html';
+import entities from 'entities';
+import csvStringify from 'csv-stringify';
+
+import title from '../../util/title';
+import FieldContext from './FieldContext';
+
+const forceDownload = (filename, csv) => {
+    const fakeLink = document.createElement('a');
+    document.body.appendChild(fakeLink);
+
+    fakeLink.setAttribute('href', `data:application/octet-stream;charset=utf-8,${encodeURIComponent(csv)}`);
+    fakeLink.setAttribute('download', `${filename}.csv`);
+    fakeLink.click();
+};
+
+class Export extends Component {
+
+    static contextTypes = {
+        store: React.PropTypes.object,
+        muiTheme: React.PropTypes.object,
+    };
+
+    componentDidMount() {
+        this.generateCsvDebounced();
+    }
+
+    generateCsv() {
+        const { resource, data, ids, children, exportOptions, onDownload, basePath } = this.props;
+        const { store, muiTheme } = this.context;
+        const csvData = [];
+        const fields = React.Children.toArray(children);
+
+        csvData.push(fields.map(field => title(field.props.label, field.props.source)));
+
+        ids.forEach(id => csvData.push(
+            fields.map((field) => {
+                const cellContent = ReactDOMServer.renderToStaticMarkup(
```

Comment:
Instead of using `ReactDomServer`, can't you render the CSV inside an invisible `<code>` tag, and then read the `innerHtml` to grab the `csvData`? This would remove the need for the context passing component.